src/charm.py

```diff
@@ -20,48 +22,30 @@ def __init__(self, *args):
 
         self.log = logging.getLogger()
 
-        if not self.model.unit.is_leader():
-            self.log.info("Not a leader, skipping set_pod_spec")
-            self.model.unit.status = ActiveStatus()
-            return
-
         self.image = OCIImageResource(self, "oci-image")
 
-        try:
-            self.interfaces = get_interfaces(self)
-        except NoVersionsListed as err:
-            self.model.unit.status = WaitingStatus(str(err))
-            return
-        except NoCompatibleVersions as err:
-            self.model.unit.status = BlockedStatus(str(err))
-            return
-        else:
-            self.model.unit.status = ActiveStatus()
-
         self.framework.observe(self.on.install, self.set_pod_spec)
         self.framework.observe(self.on.upgrade_charm, self.set_pod_spec)
         self.framework.observe(self.on.config_changed, self.set_pod_spec)
         self.framework.observe(self.on["mysql"].relation_changed, self.set_pod_spec)
-
```

Comment:
Shall we keep the naming consistent among charms and change `self.set_pod_spec` to `main`?

Reply:
Also, any particular reason why you didn't use the `for event in...`